marvin-ai/marvin-python-toolbox
# Adding support to build docker images with the engine-httpserver running as default

Now we have a CLI to build docker images.

This fixes issue #35 

This first version is able to build two type of docker images:
 - base - This is the base docker image that should be extended by any other marvin image.
 - spark - This is a docker image that uses marvin base and adds spark on top of it. Ideally should be used by engines that are spark-dependent.

Requirements:
 - The host machine should have docker installed.
 - The user running marvin CLI should be part of docker group (to avoid running with sudo). See https://docs.docker.com/engine/installation/linux/linux-postinstall/

How to test:
 - Clone this branch
 - Generate a new project: marvin engine-generate
 - Customize with your code (optional)
 - Build the docker image: marvin engine-dockerbuild (run --help and play around with parameters)
 - Run the image just build: docker run -it marvinai/marvin-spark
 - Call the REST pipeline endpoint and check that everything is working fine.

Custom spark configuration:
 - If you want to use custom spark files, replace the base files in <engine_home>/docker/marvin-spark-docker/spark-conf/* and build the image.
